src/js/components/DataChart/DataChart.js

```diff
@@ -419,16 +445,19 @@ const DataChart = forwardRef(
     // except when using offset, then add even more horizontal pad
     const pad = useMemo(() => {
       if (padProp !== undefined) return padProp;
-      const result = {};
+      let pad0;
```

Comment:
The use of `0` and `1` isn't clear to me at first. It might make more sense to just have is remain as `padHorizontal` and `padVertical` assuming direction = horizontal is default then have a code comment near line 458 about why we swap it when vertical?